torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1130,12 +1145,19 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
   if (sequenceNum_) {
     sequenceNum_->increment();
   }
+
+  // Inputs must either all be on different devices,
+  // or all be on the same device.
```

Comment:
Could you briefly describe here the effect of same vs. different device, meaning how NCCL handles these two use cases? Someone who is not familiar with NCCL (and coalesced ops) might not understand what it means to be for all tensors to be on the same device for a collective call.